Provide better error message if Node main module not found

I forgot to do a number of things before `pulumi preview` on a managed stack:
- Run `npm install`
- Run `npm link` (I'm still on 0.9.13)
- Run `npm build` to build the TypeScript

Once I saw the error message, I realized what was going on, but most users probably won't.

```
Uploading program:  27.19 KiB / 27.19 KiB [=================] 100.00% 0s
Previewing changes:
+ pulumi:pulumi:Stack: (create)
    [urn=urn:pulumi:donna-testing::url-shortener::pulumi:pulumi:Stack::url-shortener-donna-testing]
error: Running program '/tmp/update-f8341141-da4c-4e43-9c6a-9a4ff0096ddd490789845/contents' failed with an unhandled exception:
error: { Error: Cannot find module '/tmp/update-f8341141-da4c-4e43-9c6a-9a4ff0096ddd490789845/contents'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at runtime.runInPulumiStack (/opt/pulumi/node_modules/pulumi/cmd/run/index.js:140:16)
    at new Stack (/opt/pulumi/node_modules/pulumi/runtime/stack.js:30:23)
    at Object.runInPulumiStack (/opt/pulumi/node_modules/pulumi/runtime/stack.js:18:15)
    at main (/opt/pulumi/node_modules/pulumi/cmd/run/index.js:132:13)
    at Object.<anonymous> (/opt/pulumi/node_modules/pulumi/cmd/run/index.js:144:1)
    at Module._compile (module.js:570:32) code: 'MODULE_NOT_FOUND' }
An error occurred while advancing the preview: an unhandled error occurred: Program exited with non-zero exit code: 1
error: preview unsuccessful: status failed
```

--- a/sdk/nodejs/cmd/run/index.ts
+++ b/sdk/nodejs/cmd/run/index.ts
@@ -128,7 +128,8 @@ function reportModuleLoadFailure(program: string, error: Error): never {
         }
     }
 
-    console.error(`  * We're not really sure! Here's the exception message we received: ${error.message}`);
+    console.error("  * Yowzas, our sincere apologies, we haven't seen this before!");
+    console.error(`    Here is the raw exception message we received: ${error.message}`);
     return process.exit(1);
 }
 

